model/trigger/task.go

```diff
@@ -338,28 +352,24 @@ func (t *taskTriggers) taskRegression(sub *event.Subscription) (*notification.No
 				}
 				grip.Warning(errMessage)
 				if !shouldSend {
-					return nil, nil
+					return shouldNotify, nil, nil
 				}
 
 			} else {
 				var old bool
 				if old, err = taskFinishedTwoOrMoreDaysAgo(lastAlerted.TaskId); !old {
-					return nil, errors.Wrap(err, "failed to process regression trigger")
+					return shouldNotify, nil, errors.Wrap(err, "failed to process regression trigger")
 				}
 			}
 		}
 	}
+	shouldNotify = true
 
-	n, err := t.generate(sub)
-	if err != nil {
-		return nil, err
-	}
-
-	rec := newAlertRecord(t.task, alertrecord.TaskFailTransitionId)
+	rec := newAlertRecord(t, alertrecord.TaskFailTransitionId)
```

Comment:
You need to lift this statement out of this function, because it will break the task-regression trigger.

Reply:
Could you elaborate? This just creates a struct and returns it without any other side effects

Reply:
Oh you're right, I thought it was inserting too. Nevertheless, it'd probably be better to simplify the return type of this function, and make it only do the regression verification. Let's just have the appropriate callers deal with creation of alert records.

Reply:
The signature is definitely not great, but I think doing that will actually be messier since it would need to return the previous task so that the caller knows what order # to put in the alert

Reply:
The caller is passing in a task `t`, so they have the task and thus the order number already. If the record needs to be created, `newAlertRecord(t, alertrecord.TaskFailTransitionId)`. How would it make things messier?

Further more, why does a function called isTaskRegression do anything with alert record in the first place?

Reply:
The order is from the previous completed task which needs to be looked up. I can have the caller pass the previous task in as a param - would you prefer that?

Reply:
oh i'm sorry. I repeatedly skipped over the block just below this.